src/app-layer-mqtt.c

```diff
@@ -37,20 +31,9 @@
 void RegisterMQTTParsers(void)
 {
     SCLogDebug("Registering Rust mqtt parser.");
-    uint32_t max_msg_len = 1048576; /* default: 1MB */
 
-    if (AppLayerParserConfParserEnabled("tcp", "mqtt")) {
-        ConfNode *p = ConfGetNode("app-layer.protocols.mqtt.max-msg-length");
-        if (p != NULL) {
-            uint32_t value;
-            if (ParseSizeStringU32(p->val, &value) < 0) {
-                SCLogError("invalid value for max-msg-length: %s", p->val);
-            } else {
-                max_msg_len = value;
-            }
-        }
-        rs_mqtt_register_parser(max_msg_len);
-    }
+    rs_mqtt_register_parser();
```

Comment:
I think we can go one step further and get rid of this C file since there are no unit tests:
- rename `rs_mqtt_register_parser` to something more library friendly, like `SCMqttRegisterParser`
- call that where `RegisterMQTTParsers` was previously called 
- remove the C wrapper around the Rust function call.

Reply:
And let's remove src/app-layer-mqtt.c and h file to increase rust code proportion :-)